src/hotspot/os/windows/os_windows.cpp

```diff
@@ -1305,18 +1305,19 @@ void os::check_dump_limit(char* buffer, size_t buffsz) {
     const char* cwd = get_current_directory(nullptr, 0);
     int pid = current_process_id();
     if (cwd != nullptr) {
-      jio_snprintf(buffer, buffsz, "%s\\hs_err_pid%u.mdmp", cwd, pid);
+      jio_snprintf(buffer, bufferSize, "%s\\hs_err_pid%u.mdmp", cwd, pid);
     } else {
-      jio_snprintf(buffer, buffsz, ".\\hs_err_pid%u.mdmp", pid);
+      jio_snprintf(buffer, bufferSize, ".\\hs_err_pid%u.mdmp", pid);
     }
 
-    if (dumpFile == nullptr &&
+    if (check_only == false && dumpFile == nullptr &&
        (dumpFile = CreateFile(buffer, GENERIC_WRITE, 0, nullptr, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr))
                  == INVALID_HANDLE_VALUE) {
-      jio_snprintf(buffer, buffsz, "Failed to create minidump file (0x%x).", GetLastError());
+      jio_snprintf(buffer, bufferSize, "Failed to create minidump file (0x%x).", GetLastError());
       status = false;
     }
   }
+
   VMError::record_coredump_status(buffer, status);
```

Comment:
Again I think this was intended to be conditional on `check_only`.

Reply:
Fixed.